File: src/main/java/com/java/scogen/dataGenerator/EmployeeDeptDataset.java
package com.java.scogen.dataGenerator;

import com.java.scogen.pojo.EmployeeDeptPojo;
import com.java.scogen.constants.DeptID;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

import java.util.Arrays;
import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;

public class EmployeeDeptDataset {

    private SparkSession spark;

    public EmployeeDeptDataset(SparkSession spark) {
        this.spark= spark;
    }

    public Dataset<Row> empDeptDataset(List<String> empid)
    {
        List<String> deptlist = Arrays.stream(DeptID.values()).map(Enum::toString)
                .collect(Collectors.toList());
        List<EmployeeDeptPojo>firstDeptList = empid.stream().limit(500)
                .map(x-> new EmployeeDeptPojo(x,deptlist.get(new Random().nextInt(2))))
                .collect(Collectors.toList());

        List<EmployeeDeptPojo>secondDeptList = empid.stream().skip(500)
                .map(x-> new EmployeeDeptPojo(x,deptlist.get(new Random().nextInt(3)+2)))
                .collect(Collectors.toList());

        return spark.createDataFrame(firstDeptList,EmployeeDeptPojo.class)
                .union(spark.createDataFrame(secondDeptList,EmployeeDeptPojo.class));
    }
}

File: src/main/scala/com/scala/scogen/datagenerator/EmployeeDeptDataset.scala
package com.scala.scogen.datagenerator

import com.scala.scogen.caseclasses.EmpDeptPojo
import com.scala.scogen.constants.DeptID
import org.apache.spark.sql.{Dataset, Row, SparkSession}

import scala.util.Random

class EmployeeDeptDataset {

  def empDeptDataset(spark:SparkSession,empid: List[String]):Dataset[Row] = {

    import spark.implicits._
    val deptFirstList = DeptID.deptIds.slice(0,2).map(_._1)
    val deptSecondtList = DeptID.deptIds.slice(2,5).map(_._1)
    val FirstSetOfEmp = empid.slice(0,500).map(eid => EmpDeptPojo(eid,deptFirstList(new Random().nextInt(2))))
    val SecondSetOfEmp = empid.slice(500,1000).map(eid => EmpDeptPojo(eid,deptSecondtList(new Random().nextInt(3))))

    spark.sparkContext.parallelize(FirstSetOfEmp ++ SecondSetOfEmp).toDF()
  }
}
